src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -219,6 +234,9 @@ def execute(self, reqStatus):
                     msg = "\tError: %s" % str(ex)
                     self.logger.exception(msg)
                 if success:
+                    # clean-up local persistent storage if move operation was successful
+                    if wflow.dataReplacement:
```

Comment:
I think we should move these 2 lines only after persisting the new rules in the transfer document, line:
```
self.createTransferDoc(wflow.getName(), transfers)
```
below.

Reply:
In addition - for the block below - we cannot execute this status transition:
```
self.change(wflow.getName(), 'staging', self.__class__.__name__)
```

for workflows under data replacement.

Reply:
ok